### ISSUE TITLE
Unexpected error when creating a Dict from a list of lists len 2

### ISSUE THREAD

**Issue by adamtheturtle** [python/mypy#6542]
I have the following code which works.

```python
from typing import Dict

my_list = ['a', 'b']
my_dict: Dict[str, str] = dict([my_list])
print(my_dict)  # {'a': 'b'}
```

``mypy`` gives the following error:

```
example.py:4: error: List item 0 has incompatible type "List[str]"; expected "Tuple[str, str]"
```

I expect that in this case ``mypy`` would not show an error because the code is valid.

From https://docs.python.org/3/library/stdtypes.html#dict:

> Each item in the iterable must itself be an iterable with exactly two objects.

**Comment by srittau** [comment]
Seems like a typeshed issue. Possibly `dict.__init__()` is missing an overload like:

```python
    @overload
    def __init__(self, iterable: Iterable[Iterable[Union[_KT, _VT]]], **kwargs: _VT) -> None: ...
```

**Comment by adamtheturtle** [comment]
@srittau I'm not yet good a assessing which of these repos issues belong to! I will create a type shed issue, thank you.

**Comment by adamtheturtle** [comment]
Closing in favour of https://github.com/python/typeshed/issues/2862.
